scheduledJobs/giveApples.js

```
@@ -0,0 +1,88 @@
+const cron = require('node-cron');
+const Promise = require('promise');
+const { fetchAllMessagesByChannel } = require('../common/discordutil');
+
+const logbookChannelNameStem = 'class-n-club-logbook';
+
+// TODO: need to get the apple role ids for BA server
+const firstRoleId = '';
+const secondRoleId = '';
+
+
+module.exports = {
+  name: 'giveApples',
+  async start(client) {
+    // running job at 02:00 at America/Chicago timezone
+    cron.schedule('0 2 * * *', async () => {
+      // gather logbook channels
+      const logbookChannels = client.channels.cache.filter(c => c.name.toLowerCase().includes(logbookChannelNameStem));
+      const userIdsInLogbook = [];
+      const membersNeedingApples = [];
+      const messagesWithAppliesApplied = [];
+
+      const addAppleRoles = () => {
+        const bothRoles = [firstRoleId, secondRoleId];
+
+        // Counts number of times a user appears in ids
+        const userFrequency = {};
+        for (let i = 0; i < userIdsInLogbook.length; i++) {
+          userFrequency[userIdsInLogbook[i]] = userFrequency[userIdsInLogbook[i]] ? userFrequency[userIdsInLogbook[i]] + 1 : 1;
+        }
+
+        // Adds roles based on userFrequency counts
+        membersNeedingApples.forEach((member) => {
+          if (userFrequency[member.id] >= 2) {
+            member.roles.add(bothRoles);
+          } else if (!member.roles.cache.has(firstRoleId)) {
+            member.roles.add([firstRoleId]);
+          } else {
+            member.roles.add([secondRoleId]);
+          }
+        });
+      };
+
+      if (logbookChannels) {
```

Can we throw an error if for some reason logbookChannels ends up falsy